src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs

```diff
@@ -107,7 +124,11 @@ public EncryptionKey? EncryptionKey
         /// </summary>
         public DatabaseConfiguration()
         {
-
+#if !MACCATALYST
+            if(!RuntimeInformation.IsOSPlatform(OSPlatform.OSX)) {
+                _mmapEnabled = Constants.DefaultDatabaseMmapEnabled;
+            }
+#endif
```

Comment:
From the CBL Discussion meeting, we don't need to implementation any guardrails or special logic for macOS. LiteCore will decide what to do. After all, this is minor but I am wondering why not including MACCATYLYST?